src/lib.rs

```diff
@@ -1692,6 +1692,20 @@ impl Bump {
         unsafe { footer.as_ref().allocated_bytes }
     }
 
+    /// Calculates the number of bytes requested from the Rust allocator for this `Bump`.
+    ///
+    /// This number is equal to the [`allocated_bytes()`](Self::allocated_bytes) plus
+    /// the size of the chunk headers.
+    pub fn allocated_bytes_with_headers(&self) -> usize {
+        let headers_size = unsafe {
+            self.iter_allocated_chunks_raw()
+                .map(|(_, size)| size)
+                .sum::<usize>()
+                * mem::size_of::<ChunkFooter>()
```

Comment:
The first three lines are summing the total byte size of the chunks, and then the fourth line is multiplying that by the chunk footer size. This is not correct, it should be counting the number of chunks and multiplying that by the footer size. As written, this will vastly over-report memory.

Should be something like this instead:

```suggestion
            self.iter_allocated_chunks_raw().count()
                * mem::size_of::<ChunkFooter>()
```

Reply:
My bad. Thanks!